### PR TITLE
fix(core): show navigation button on mouse hover and fix item order for looped carousel

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
Closes SAP/fundamental-ngx#4432
Closes SAP/fundamental-ngx#4434

#### Please provide a brief summary of this pull request.
This PR fixes following
1. #4434 For carousel with navigation button inside content area, navigation button will be visible on mouse hover.
2. #4432 For looped carousel, when previous button is pressed first time before clicking other button, item order was not correct after that. this is fixed now.
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [NA] update `README.md`
- [NA] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by kavya-b — APPROVED** [review]
LGTM

**Review by Betrozov — APPROVED** [review]

**Review by KevinOkamoto — APPROVED** [review]

**Comment by maxym-valor** [comment]
Approved!

File: libs/core/src/lib/carousel/carousel.component.scss [hunk 1]
```diff
@@ -4,6 +4,14 @@ fd-carousel:focus {
  outline: 1px dotted;
 }
 
+.fd-carousel:hover .fd-carousel__content .fd-button.fd-carousel__button{
```
Comment (N1XUS) [inline libs/core/src/lib/carousel/carousel.component.scss:7]: Code formatting. Please use SCSS nesting

Comment (DeepakSap14) [reply libs/core/src/lib/carousel/carousel.component.scss]: done.

File: libs/core/src/lib/carousel/carousel.component.scss [hunk 2]
```diff
@@ -4,6 +4,26 @@ fd-carousel:focus {
...
+
+.fd-carousel:hover {
+  .fd-carousel__content {
+    .fd-carousel__button {
+      &.fd-button {
+        display: block;
+      }
+    }
+  }
+}
```
Comment (InnaAtanasova) [inline libs/core/src/lib/carousel/carousel.component.scss:25]: This should be in fundamental-styles

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
We are no longer approving PRs with css that belongs to Fundamental-styles.

File: libs/core/src/lib/carousel/carousel.component.scss (hunk 2 above)
Comment (DeepakSap14) [reply libs/core/src/lib/carousel/carousel.component.scss:25]: have raised https://github.com/SAP/fundamental-styles/pull/2431 in styles.

**Comment by droshev** [comment]
Closing old PRs - feel free to reopen it if/when needed

**Comment by DeepakSap14** [comment]
will reopen once fundamental-styles [PR#2431](https://github.com/SAP/fundamental-styles/pull/2431) is merged and available in Ngx.

**Pull request closed without merging**